### DISCUSSION TITLE
addons load relation throw error when the foreign key is null

### DISCUSSION THREAD

**Discussion by farshadfahimi** [adonisjs/core#3376] [Ideas]
I have simple query like below
`let user: User | null = await order.related('user').query().debug(true).first()`

some times the `usersId` is null in my query and the above query on execute got below error
`Cannot preload \"user\", value of \"Order.usersId\" is undefined`

I think it's not a good Idea to thorw an error it's better to return null

**Comment by thetutlage** [comment]
You can simply avoid running the query by checking for the `userId` beforehand.

```ts
let user: User | null = order.userId ? await order.related('user').query().debug(true).first() : null


```
